compiler/rustc_metadata/src/creader.rs

```diff
@@ -1093,3 +1093,42 @@ fn alloc_error_handler_spans(sess: &Session, krate: &ast::Crate) -> Vec<Span> {
     visit::walk_crate(&mut f, krate);
     f.spans
 }
+
+// On Windows the compiler would sometimes intermittently fail to open the
+// proc-macro DLL with `Error::LoadLibraryExW`. It is suspected that something in the
+// system still holds a lock on the file, so we retry a few times before calling it
+// an error.
+fn load_dylib(path: &Path, max_attempts: usize) -> Result<libloading::Library, String> {
+    assert!(max_attempts > 0);
+    let mut attempt = 0;
+
+    loop {
```

Comment:
I think it would be clearer to write this as a `for attempt in 0..max_attempts {` type loop. You can then make returning the `Err` on 1129 the last expression in this function.

Reply:
Yes, that is easier to read. Thanks for the feedback!